Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -76,218 +90,239 @@ + (BNCPreferenceHelper *)getInstance {
-    [BNCPreferenceHelper writeObjectToDefaults:KEY_LINK_CLICK_ID value:linkClickId];
-}
+- (NSString *)getLinkClickIdentifier {
+    if (!_linkClickIdentifier) {
+        _linkClickIdentifier = [self readStringFromDefaults:BRANCH_PREFS_KEY_LINK_CLICK_IDENTIFIER];
+    }
 
-+ (NSString *)getLinkClickID {
-    return [BNCPreferenceHelper readStringFromDefaults:KEY_LINK_CLICK_ID];
+    return _linkClickIdentifier;
 }
 
-+ (void)setSessionParams:(NSString *)sessionParams {
-    [BNCPreferenceHelper writeObjectToDefaults:KEY_SESSION_PARAMS value:sessionParams];
+- (void)setLinkClickIdentifier:(NSString *)linkClickIdentifier {
+    if (![_linkClickIdentifier isEqualToString:linkClickIdentifier]) {
+        _linkClickIdentifier = linkClickIdentifier;
+        [self writeObjectToDefaults:BRANCH_PREFS_KEY_LINK_CLICK_IDENTIFIER value:linkClickIdentifier];
+    }
 }
 
-+ (NSString *)getSessionParams {
-    return [BNCPreferenceHelper readStringFromDefaults:KEY_SESSION_PARAMS];
+- (NSString *)getSessionParams {
+    if (_sessionParams) {
+        _sessionParams = [self readStringFromDefaults:BRANCH_PREFS_KEY_SESSION_PARAMS];
+    }
+    
+    return _sessionParams;
 }
 
-+ (void)setInstallParams:(NSString *)installParams {
-    [BNCPreferenceHelper writeObjectToDefaults:KEY_INSTALL_PARAMS value:installParams];
+- (void)setSessionParams:(NSString *)sessionParams {
+    if (![_sessionParams isEqualToString:sessionParams]) {
+        _sessionParams = sessionParams;
+        [self writeObjectToDefaults:BRANCH_PREFS_KEY_SESSION_PARAMS value:sessionParams];
+    }
 }
 
-+ (NSString *)getInstallParams {
-    return [BNCPreferenceHelper readStringFromDefaults:KEY_INSTALL_PARAMS];
+- (NSString *)getInstallParams {
+    if (!_installParams) {
+        _installParams = [self readStringFromDefaults:BRANCH_PREFS_KEY_INSTALL_PARAMS];
+    }
+    
+    return _installParams;
 }
 
-+ (void)setUserURL:(NSString *)userUrl {
-    [BNCPreferenceHelper writeObjectToDefaults:KEY_USER_URL value:userUrl];
+- (void)setInstallParams:(NSString *)installParams {
+    if (![_installParams isEqualToString:installParams]) {
+        _installParams = installParams;
+        [self writeObjectToDefaults:BRANCH_PREFS_KEY_INSTALL_PARAMS value:installParams];
+    }
 }
 
-+ (NSString *)getUserURL {
-    return [BNCPreferenceHelper readStringFromDefaults:KEY_USER_URL];
+- (NSString *)getUserURL {
+    if (!_userUrl) {
+        _userUrl = [self readStringFromDefaults:BRANCH_PREFS_KEY_USER_URL];
+    }
+    
+    return _userUrl;
 }
 
-+ (BOOL)getIsReferrable {
-    return [BNCPreferenceHelper readBoolFromDefaults:KEY_IS_REFERRABLE];
+- (void)setUserURL:(NSString *)userUrl {
+    if (![_userUrl isEqualToString:userUrl]) {
+        _userUrl = userUrl;
+        [self writeObjectToDefaults:BRANCH_PREFS_KEY_USER_URL value:userUrl];
+    }
 }
 
-+ (void)setIsReferrable {
-    [BNCPreferenceHelper writeBoolToDefaults:KEY_IS_REFERRABLE value:YES];
+- (BOOL)getIsReferrable {
+    return _isReferrable;
```

Comment:
It is... but because it's a boolean, it can't be "lazy loaded" like all the objects in here. We instead have to load it [at initialization time](https://github.com/BranchMetrics/Branch-iOS-SDK/pull/134/files#diff-4db16133408e16908e5c2f5bfa6038e5R73) :(